src/main/java/com/bootcamp/demo/repository/FirebaseUserRepository.java

```diff
@@ -0,0 +1,84 @@
+package com.bootcamp.demo.repository;
+
+import com.bootcamp.demo.UserBuilder;
+import com.bootcamp.demo.model.User;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class FirebaseUserRepository implements UserRepository{
+    Firestore firestoreDB;
+
+    public FirebaseUserRepository(Firestore firestoreDB) {
+        this.firestoreDB = firestoreDB;
+    }
+
+    @Override
+    public Set<User> findAll() {
+        try {
+        Iterable<QueryDocumentSnapshot> users = firestoreDB.collection("accounts")
+                .get()
+                .get()
+                .getDocuments();
+
+            return StreamSupport
+                    .stream(users.spliterator(), false)
+                    .map(UserBuilder::documentToModel)
+                    .collect(Collectors.toSet());
+        } catch (ExecutionException | InterruptedException e) {
+            throw new RuntimeException();
+        }
+    }
+
+    @Override
+    public String save(User user)  {
+        try {
+            ApiFuture<WriteResult> collectionFuture = firestoreDB
+                    .collection("accounts")
+                    .document("insert_id_here") //here an Id model would be nice
+                    .set(user);
+            return collectionFuture.get().getUpdateTime().toString();
+        } catch (ExecutionException | InterruptedException e) {
+             throw new RuntimeException();
+        }
+    }
+
+    @Override
+    public void remove(User user) {
+        ApiFuture<WriteResult> writeResult = firestoreDB.collection("accounts")
+                .document("insert_id_here") //here an Id model would be nice
+                .delete();
+    }
+
+    @Override
+    public User findById(String id)  {
```

Comment:
Am putea implementa oare si un findOne(String email, String password) ca sa o folosim pentru verificarea  la LogIn a existentei acestui User?

Reply:
M-am gandit la un approach in care nu facem 'search' dupa password la nivel de repository ci doar dupa email, facem un 'load' la user cu email-ul respectiv la nivel de service. Si tot la service sa ne folosim de AuthenticationManager ca sa ne verifice user-ul

In general am lucrat cu metode de e.g. :
` findByEmail(String email);
 existsByEmail(String email );`

Reply:
Ok, sounds good, eu zic ca putem continua atunci cu abordarea asta.